### PR TITLE
[SPARK-26076][Build][Minor] Revise ambiguous error message from load-spark-env.sh

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?

When I try to run scripts (e.g. `start-master.sh`/`start-history-server.sh ` in latest master, I got such error:
```
Presence of build for multiple Scala versions detected.
Either clean one of them or, export SPARK_SCALA_VERSION in spark-env.sh.
```

The error message is quite confusing. Without reading `load-spark-env.sh`,  I didn't know which directory to remove, or where to find and edit the `spark-evn.sh`.

This PR is to make the error message more clear. Also change the script for less maintenance when we add or drop Scala versions in the future.
As now with https://github.com/apache/spark/pull/22967, we can revise the error message as following(in my local setup):

```
Presence of build for multiple Scala versions detected (/Users/gengliangwang/IdeaProjects/spark/assembly/target/scala-2.12 and /Users/gengliangwang/IdeaProjects/spark/assembly/target/scala-2.11).
Remove one of them or, export SPARK_SCALA_VERSION=2.12 in /Users/gengliangwang/IdeaProjects/spark/conf/spark-env.sh.
Visit https://spark.apache.org/docs/latest/configuration.html#environment-variables for more details about setting environment variables in spark-env.sh.
```

## How was this patch tested?

Manual test

### CODE REVIEW

**Comment by gengliangwang** [comment]
@srowen

File: bin/load-spark-env.sh [hunk 1]
```diff
@@ -47,8 +47,8 @@ if [ -z "$SPARK_SCALA_VERSION" ]; then
   ASSEMBLY_DIR1="${SPARK_HOME}/assembly/target/scala-2.12"
 
   if [[ -d "$ASSEMBLY_DIR2" && -d "$ASSEMBLY_DIR1" ]]; then
-    echo -e "Presence of build for multiple Scala versions detected." 1>&2
-    echo -e 'Either clean one of them or, export SPARK_SCALA_VERSION in spark-env.sh.' 1>&2
+    echo -e "Presence of build for both scala versions(SCALA 2.11 and SCALA 2.12) detected." 1>&2
```
Comment (srowen) [inline bin/load-spark-env.sh]: How about something like "Multiple Scala versions detected ($ASSEMBLY_DIR1 and $ASSEMBLY_DIR2). Remove one, or export SPARK_SCALA_VERSION=(version) in load-spark-env.sh"

That adds more info and maybe is less maintenance as we add or drop Scala versions.

Comment (gengliangwang) [reply bin/load-spark-env.sh]: Make sense. Let me have a quick update

File: bin/load-spark-env.sh [hunk 2]
```diff
@@ -42,19 +42,20 @@ fi
...
-  if [[ -d "$ASSEMBLY_DIR2" && -d "$ASSEMBLY_DIR1" ]]; then
-    echo -e "Presence of build for multiple Scala versions detected." 1>&2
-    echo -e 'Either clean one of them or, export SPARK_SCALA_VERSION in spark-env.sh.' 1>&2
+  SCALA_VERSION1=2.12
+  SCALA_VERSION2=2.11
+
+  ASSEMBLY_DIR1="${SPARK_HOME}/assembly/target/scala-${SCALA_VERSION1}"
+  ASSEMBLY_DIR2="${SPARK_HOME}/assembly/target/scala-${SCALA_VERSION2}"
+  if [[ -d "$ASSEMBLY_DIR1" && -d "$ASSEMBLY_DIR2" ]]; then
+    echo -e "Presence of build for multiple Scala versions detected($ASSEMBLY_DIR1 and $ASSEMBLY_DIR2)." 1>&2
```
Comment (srowen) [inline bin/load-spark-env.sh]: Nit: space after 'detected'. I think you can put this in one string if you like.

**Comment by vanzin** [comment]
The reason why it mentioned `spark-env-sh` is because that is a script that is loaded by `load-spark-env.sh` and is user-defined. From that script:

```
  if [ -f "${SPARK_CONF_DIR}/spark-env.sh" ]; then
    # Promote all variable declarations to environment (exported) variables
    set -a
    . "${SPARK_CONF_DIR}/spark-env.sh"
    set +a
  fi
```

Users shouldn't modify `load-spark-env.sh`, so your updated message is actually wrong.

**Comment by gengliangwang** [comment]
Hi @vanzin ,
thanks for pointing it out! I have updated the script and PR description.

**Comment by gengliangwang** [comment]
retest this please.

**Comment by vanzin** [comment]
I'm not sure this is actually helping much or making things less confusing.

If `spark-env.sh` doesn't exist, your message will tell people to modify a file that doesn't exist.

If it's triggered in a subsequent invocation of this script, where the file has already been loaded, it will just say `spark-env.sh` without its location, like before. So nothing is gained.

`spark-env.sh` is actually documented in Spark's configuration doc. So if you want to point people at the doc instead, I could see that as an improvement. But you current change doesn't really change anything from before to me.

**Comment by gengliangwang** [comment]
@vanzin I see your point. I will add a link to https://spark.apache.org/docs/latest/configuration.html. Thanks for the suggestion.
In my case, I didn't know where to find or edit `spark-env.sh` at that time. I tried run `find . -name spark-env.sh` and got nothing.
The script will only print `spark-env.sh` without its location only if SPARK_ENV_LOADED is not set.

**Comment by gengliangwang** [comment]
retest this please.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**